fix(ui): Span tree connector design

vuluongj20 opened the pull request:
The current color for span tree connectors is `gray200`. This is too light when the line width is just `1px`. On the other hand, `gray300` is too dark (it's meant as an accessible text color). A way to solve this is by setting the line width at 2px while keeping the color at `gray200`.

Before:
<img width="469" alt="Screen Shot 2021-12-07 at 6 38 28 PM" src="https://user-images.githubusercontent.com/44172267/145139502-5622b3df-8c69-4785-902c-c690bc893d91.png">
<img width="469" alt="Screen Shot 2021-12-07 at 6 46 46 PM" src="https://user-images.githubusercontent.com/44172267/145139504-8ffebc20-4154-4f6e-a9fb-ff106bc63e42.png">

After:
<img width="561" alt="Screen Shot 2021-12-09 at 2 41 04 PM" src="https://user-images.githubusercontent.com/44172267/145487657-7e6af8b4-7b9d-4342-9fe9-487f2491752a.png">
<img width="561" alt="Screen Shot 2021-12-09 at 2 41 18 PM" src="https://user-images.githubusercontent.com/44172267/145487659-7af186f7-d1e9-4b09-857f-f52ce5b65016.png">

vuluongj20 committed:
fix(ui): Span tree connector color

The current color for span tree connectors is gray200. This is too light. Instead, we should use gray300 at 0.4 opacity.

vuluongj20 commented on static/app/components/performance/waterfall/treeConnector.tsx:
```diff
@@ -14,9 +14,10 @@ const TREE_TOGGLE_CONTAINER_WIDTH = 40;
+  border-left: 1px ${p => (p.orphanBranch ? 'dashed' : 'solid')} ${p => p.theme.subText};
   top: -5px;
   position: absolute;
+  opacity: 0.4;
```
@dashed Only added opacity because `gray300` seems too strong as line color (sadly we don't have a color between `gray200` and `gray300`). An alternative is to use the color package, and set the border color to `Color(theme.gray300).alpha(0.4).string()`. Wdyt?

dashed replied:
@vuluongj20 Yes, the generated colour string (without the alpha) would be preferred.

evanpurkhiser replied:
Let's make sure it doesn't run a huge number of times if we use that computation.

dashed replied:
Ideally, we'd precompute it and hardcode it, instead of during runtime.

vuluongj20 replied:
@evanpurkhiser @dashed imo, neither `gray200` nor `gray300` works well at full opacity. Here's another solution: we can keep the line color at `gray200`, but increase the line width to 2px. This way the lines are more prominent (thus addressing the original problem) but not too contrasty. Dm'ed Dora and she's cool w/ it. What do y'all think?

<img width="561" alt="Screen Shot 2021-12-09 at 2 41 18 PM" src="https://user-images.githubusercontent.com/44172267/145487659-7af186f7-d1e9-4b09-857f-f52ce5b65016.png">

<img width="561" alt="Screen Shot 2021-12-09 at 2 41 04 PM" src="https://user-images.githubusercontent.com/44172267/145487657-7e6af8b4-7b9d-4342-9fe9-487f2491752a.png">

dashed replied:
@vuluongj20 Works for me. I'll defer to @getsentry/performance team for any in-depth review with regards to the new changes.

dashed replied:
@vuluongj20 Since the tree branches have a wider width. Would you be willing to explore making use of an svg for the dashed branches?

Also did you manually set the dashed lines for the new screenshots? If not, that should not occur and is a bug.

vuluongj20 replied:
@dashed Hmm why would we want to use an svg for dashed branches?

And yes, I manually set the dashed lines, to illustrate what it would look like when a connector line has type `orphan`.

dashed replied:
@vuluongj20 The dashed branches are inconsistent across browsers since they each have different opinions of what `border-style: dashed;`.  In most cases they do not pattern well. 

It's been on my backlog since last year 😄 

I might likely look into it sometime next year.

vuluongj20 committed:
ref(ui): Increase line width for span tree connectors

The current line color is gray200. This is too light when the line width is 1px. Increasing the width to 2px makes the line more prominent.

Zylphrex reviewed (commented):
@vuluongj20 There seems to be some misalignment specifically around the auto grouped spans (see the blue pills in the attached screenshot).

![image](https://user-images.githubusercontent.com/10239353/145627940-8d8ba9de-1698-4750-853a-3fd4915c7c23.png)

vuluongj20 committed:
fix(ui): spanGroupBar offset

There is a slight offset on spanGroupBar (see PR #30498). This commit fixes that.

vuluongj20 committed:
Merge branch 'master' into vuluong/span-tree-connector-color

vuluongj20 committed:
fix(ui): TreeConnector offset by 1px

vuluongj20 committed:
fix(ui): Fix TransactionBar connector offset

vuluongj20 committed:
Merge branch 'master' into vuluong/span-tree-connector-color

vuluongj20 committed:
fix(ui): Fix TransactionBar connector offset

vuluongj20 committed:
Merge branch 'master' into vuluong/span-tree-connector-color
